Cython/Debugger/Cygdb.py

```diff
@@ -45,17 +45,22 @@ def make_command_file(path_to_debug_info, prefix_code='', no_import=False):
             set print pretty on
 
             python
-            # Activate virtualenv, if we were launched from one
-            import os
-            virtualenv = os.getenv('VIRTUAL_ENV')
-            if virtualenv:
-                path_to_activate_this_py = os.path.join(virtualenv, 'bin', 'activate_this.py')
-                print("gdb command file: Activating virtualenv: %s; path_to_activate_this_py: %s" % (
-                    virtualenv, path_to_activate_this_py))
-                with open(path_to_activate_this_py) as f:
-                    exec(f.read(), dict(__file__=path_to_activate_this_py))
-
-            from Cython.Debugger import libcython, libpython
+            from traceback import print_exc
+            try:
+                # Activate virtualenv, if we were launched from one
+                import os
+                virtualenv = os.getenv('VIRTUAL_ENV')
+                if virtualenv:
+                    path_to_activate_this_py = os.path.join(virtualenv, 'bin', 'activate_this.py')
+                    print("gdb command file: Activating virtualenv: %s; path_to_activate_this_py: %s" % (
+                        virtualenv, path_to_activate_this_py))
+                    with open(path_to_activate_this_py) as f:
+                        exec(f.read(), dict(__file__=path_to_activate_this_py))
+                from Cython.Debugger import libcython, libpython
+            except Exception as ex:
+                print("There was an error in Python code originating from the file ''' + str(__file__) + '''")
+                print("It used the Python interpreter " + str(sys.executable))
+                print_exc()
```

Comment:
Why not re-raise the exception here?

Reply:
print_exc gives a better error message than raising:
Example: I added a raise ValueErrror("test") to libpython.py
Without re-raising we get:

> There was an error in Python code originating from the file /home/volker/Sync/git/cython/Cython/Debugger/Cygdb.py
> It used the Python interpreter /usr/bin/python
> Traceback (most recent call last):
>   File "<string>", line 12, in <module>
>   File "/home/volker/Sync/git/cython/Cython/Debugger/libcython.py", line 60, in <module>
>     from Cython.Debugger import libpython
>   File "/home/volker/Sync/git/cython/Cython/Debugger/libpython.py", line 57, in <module>
>     raise ValueError("test")
> ValueError: test
> /tmp/tmpv_wmeqcm:28: Error in sourced command file:
> Undefined command: "cy".  Try "help".
> (gdb) 
With re-raising and without print_exc() we get:

> There was an error in Python code originating from the file /home/volker/Sync/git/cython/Cython/Debugger/Cygdb.py
> It used the Python interpreter /usr/bin/python
> Traceback (most recent call last):
>   File "<string>", line 18, in <module>
> ValueError: test
> /tmp/tmphlesqyrq:26: Error in sourced command file:
> Error while executing Python code.
> (gdb) 
But I just realised that exit(1) will stop gdb wich is what we want. I added a commit.